src/main/Scheduler.java

```diff
@@ -59,7 +61,7 @@ public void run() {
 					arrivalEvent = arrivalEvents.remove(0);
 					
 					for(FloorEvent fEvent : floorEvents) {
-						if(arrivalEvent.getCurrentFloor() == fEvent.getSource()) {
+						if((arrivalEvent.getCurrentFloor() == fEvent.getSource()) && fEvent.getDirection() == arrivalEvent.getDirection()) {
 							currentFloorEvent = fEvent; 
 							floorEventFlag = true; 

```

Comment:
You can recome floorEvent from the floorEvents list after line 65

Reply:
done